src/Microsoft.OData.Edm/Vocabularies/VocabularyAnnotationExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="VocabularyAnnotationExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm.Vocabularies.V1
+{
+    /// <summary>
+    /// Extension methods for the vocabulary annotations.
+    /// </summary>
+    public static class VocabularyAnnotationExtensions
+    {
+        /// <summary>
+        /// Gets the collection of string for a target annotatable.
+        /// </summary>
+        /// <param name="model">The model referenced to.</param>
+        /// <param name="target">The target annotatable to find annotation.</param>
+        /// <param name="term">The target annotatable to find annotation.</param>
+        /// <returns>Null or a collection string of qualified type name.</returns>
+        public static IEnumerable<string> GetVocabularyStringCollection(this IEdmModel model, IEdmVocabularyAnnotatable target, IEdmTerm term)
+        {
+            if (model == null || target == null || term == null)
+            {
+                return null;
+            }
+
+            IEdmVocabularyAnnotation annotation = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(target, term).FirstOrDefault();
+            if (annotation != null)
+            {
+                IEdmCollectionExpression collectionExpression = annotation.Value as IEdmCollectionExpression;
+                if (collectionExpression != null && collectionExpression.Elements != null)
+                {
+                    return collectionExpression.Elements.OfType<IEdmStringConstantExpression>().Select(e => e.Value);
+                }
+            }
+
+            return null;

```

Comment:
I would expect Enumerable.Empty rather than null for any API that returns an IEnumerable<T> when there aren't any items.

Reply:
return Enumerable.Empty<string>();

---
In reply to: [570055513](https://github.com/OData/odata.net/pull/1995#discussion_r570055513) [](ancestors = 570055513)

Reply:
+1

---
In reply to: [570055513](https://github.com/OData/odata.net/pull/1995#discussion_r570055513) [](ancestors = 570055513)